server/market/bookrouter.go

```diff
@@ -542,8 +541,8 @@ func (r *BookRouter) msgOrderBook(book *msgBook) *msgjson.OrderBook {
 // notifications.
 func (r *BookRouter) handleOrderBook(conn comms.Link, msg *msgjson.Message) *msgjson.Error {
 	sub := new(msgjson.OrderBookSubscription)
-	err := json.Unmarshal(msg.Payload, sub)
-	if err != nil {
+	err := msg.Unmarshal(&sub)
+	if err != nil || sub == nil {
```

Comment:
The zero values for `OrderBookSubscription` will be sure to be rejected anyway, so best to bail here.  Same as in `handleUnsubOrderBook` and the handlers in `OrderRouter`.